File: Algorithm/src/MeiTuan/LeetCode53.java
package MeiTuan;

public class LeetCode53 {

    public int maxSubArray(int[] nums) {

        if(nums.length == 0){
            return nums[0];
        }
        int ans = nums[0];
        int sum  = 0;

        for(int i = 0; i < nums.length; i++){
            if (sum >= 0){
                sum+=nums[i];
            }
            else {
                sum = nums[i];
            }
        }
        ans = Math.max(ans,sum);
        return  ans;
    }
}

File: top-interview-questions-easy/DP/LeetCode53.go
package DP

func maxSubArray(nums []int) int {
	ans, sum := nums[0], 0
	for i := 0; i < len(nums); i++ {
		if sum > 0 {
			sum += nums[i]
		} else {
			sum = nums[i]
		}
		ans = max(ans, sum)
	}
	return ans
}
